core/consensus/pow/src/lib.rs

```diff
@@ -205,11 +225,12 @@ impl<B: BlockT<Hash=H256>, C, Algorithm> Verifier<B> for PowVerifier<C, Algorith
 		let best_aux = PowAux::read(self.client.as_ref(), &best_hash)?;
 		let mut aux = PowAux::read(self.client.as_ref(), &parent_hash)?;
 
-		let (checked_header, difficulty, seal) = self.check_header::<B>(
+		let (checked_header, difficulty, seal) = self.check_header(
 			header,
 			BlockId::Hash(parent_hash),
 		)?;
-		aux.total_difficulty = aux.total_difficulty.saturating_add(difficulty);
+		aux.difficulty = difficulty;
+		aux.total_difficulty.add(difficulty);
```

Comment:
A small nit: my brain treats `add` as a function which takes `&self` and returns `Self`. For this case, it seems `increase(&mut self, other: Self) → ()` would be more appropriate, don't you think so?